### ISSUE TITLE
Project Timeline to Completion Created 

### ISSUE THREAD

**Issue by amoloney1** [fedora-infra/aaa-tracker#4]
This is an admin issue and should remain in the backlog as a guide for sprint planning and tracking.

# Q1 2021 Sprint Planning
#### Development
* Tune and test group assignation in Noggin is correct and users are imported correctly (Aurelien to edit post call) This issue is due to an issue with the import script rerun with missing properties in the plugin. Abompard is debugging this issue currently and is expected to resolve this in the next few days.
* * Issue was related to a group setting in IPA which was disabled so now its complete
* Complete all remaining issues around 2FA and TOTP - one feature is missing upstream but is out of the noggin team control. 
* * Some fedora apps are running off older versions of Fedora and repos need to be updated to point to the new URL
* Mitigation is to write a script for sys-admin access to email the ppl and add manually but this is not a blocker.
* Review and define where Noggin will land for the long term - is it accounts.fp.o or other? Compile a list of pros and cons for both landing areas and email the CPE list for wider feedback. Make a decision based on this discussion by Program Call on Wednesday 3rd March.
* * Noggin will have a new URL and remain on 
* SPIKE: Investigate how to turn FAS to read-only, and for how long
* * Periodic checking of FAS for issues to review activity until such a time that there is little to no activity and it is safe to turn off, then give final date of turn off

#### Documentation
* Edit CoC textbox to reflect the word 'account' in the notification for the user -- **DONE**
* Outline a deployment plan for production
* * In progress

#### CentOS
* Review needs for production migration
* * New migration date for CentOS is April - non blocker for Fedora

#### Communication Tasks
* Send email to user conflict list by 25th Feb 2021 - Aoife
* Update weekly program call doc on 24th FEB and 3rd MAR - Stephen
* Send status email to CPE mailing list on Sprint 3 achievements and a TODO on work remaining - Nils
* Update to community lists on 4th March on status of Noggin - Aurelien
* Community blog post on UI & Noggin work - Ryan **Draft DONE -- https://communityblog.fedoraproject.org/?p=9659&preview=1&_ppp=728fdcdeee**
* Announce production rollout date to community - Soft date is 19th March, but this comms will be defined at the end of this sprint
 
#### Potential Requirements/Open Questions
Please note the below are still open questions that are **in progress** and any decisions on such will be briefed into the team. This may come mid-sprint or sooner.
* Ensure Noggin can be used to capture the same data that the current FAS system does for Outreachy applications
* Currently the requirements are 
- Applicants can sign up for a FAS account
- CLA is signed
- Applicant can be added to the group they are interning with
* There are a lot more nice to haves but this will need some further review and discussion as we do not want to block production deployment on late requirements if there is potential to implement them at a later stage.

#### Development
* Complete user data migration script re-run **DONE** **suggestion to import less users came through mid re-run, but we changed how we imported users so its much faster**
* Update J-query schema to latest version in Noggin **DONE**
* If we need to re-install Free IPA - **DONE**
* Configure and merge Sigul **PR with requested changes are in place, awaiting review** **Backup plan for mitigation in place also no blocker here**
* Update features field to latest version of jQuery, and other applicable themes **DONE**
* Ensure Ansible ipa/client role and group vars/playbooks are set up for deployment **In Progress**
#### Documentation
* Update CentOS docs site to reflect Noggin user information - **Docs are DONE, they are a little more generic. added a link to them in the CentOS theme. fedora-infra/noggin#507**
  * *Open Question*: Should we request the maintenance of the images from the CentOS community if we use CentOS branding? -- **DONE the screenshots are all redone to be a little more generic**
* Rename 'Dude McPants' avatar in docs.fpo to something a little more generic please :) -- **DONE**
* Add alert box to uses on the Noggin Login (eh?!) alerting them to the need to adhere to the Fedora Code of Conduct -- **DONE in https://github.com/fedora-infra/noggin/issues/495**
  * Investigate if there is a requirement to have this notification for CentOS users also **Asked @rbowen in fedora-infra/noggin#495 but no response, CentOS doesn''t have a formal Code of Conduct, so OK to leave this one alone IMHO**
#### CentOS 
* Whatever you can do to help :)
#### Security & Admin Control
* Optimise 2FA for sys-admin functionality to increase speed **In Progress - needs a feature from SSSD which is due to be released soon but we dont have a timeframe on this yet**
* Add pkinit to docs to allow applicable certs be shipped for packages -- **DOCS DONE, just need this one merged and shipped https://pagure.io/fedora-packager/pull-request/166**
* Standardize use of TOTP tokens within Noggin **DONE**
#### Communication
* Send email to user conflict list requesting their assistance in resolving *as soon as* script has re-ran and conflicts list has been compiled **In Progress - needed re-run to be completed for an updated list and will be sent**

### Goals
* x2 reviews in in flight section to be done
* * One merged, one closed as no longer needed
* 2FA authentication for sys-admin groups
* * There has been some testing done and it is working but is very slow
* * additional work is needed on this issue to increase performance
* * Kinit (pkinit) needs to be merged into the docs and completed for this issue also to allow for shipment of packages needed
* Write script to monitor changes on accounts from its last run
* * There is a script written, but just needs to be tested once the re-run of user data script is complete
* Check with Stephen if he did all apps in CPE - both critical and not for codebase changes and then deploy then to stg to see if they work
* * Yes, however sigul is the last app to do as there was some complexity to this app but there is now changes to be made and then this will be merged
* Run a stg-wide ansible playbook to check that all the apps have the right settings
* * This will be done post re-import and re-install of Free IPA (next sprint)
* Emails - lots of emails!
    * Send one to the infra/devel list (fedora) requesting testing in stg
    * * Email sent on 8th feb requesting *no*  testing in staging as there was a re-import of accounts for more testing
* Users with email conflicts emailed requesting they pick one if they have two (AB will get the list to AM)
* * In draft waiting on the script re-run to complete and then will send the email to conflict users
* Move end-user documentation to docs.fpo
* * Done
* Security Review
    * Add security headers https://github.com/fedora-infra/noggin/issues/333
    * * Not complete, much more complexity to them with jQuery updates
    * Add justification lines to https://github.com/fedora-infra/noggin/issues/335
    * *  Done - PR submitted and merged
    * Review default config https://github.com/fedora-infra/noggin/issues/334
    * * New name suggested, waiting on cnf from security person and and then this will be closed
    * Get installation docs from dkirwan, review and post them to Noggin repo
    * * Not done yet
* Add user documentation to CentOS docs site
* * Unlikely done in this sprint, but needs to be checked with farrotin to see if there are any technicalities to take into consideration and any tweaks needed to be done to the centos users side when using noggin

### Goals
* Complete Critical App Testing with fasjson API in staging
* * Assign Access Correctly
* * * People need to be able to login and have the correct privileges assigned to work 
* Create/Update Documentation for Noggin
* * Installation guide for API for maintainers
* * How To guide for sys-admin & Troubleshooting
* * User guide for Noggin
* * Noggin Wiki
* Roll to production!

***************
# Q4 2020 Sprint Planning

### Sprint Goals
* Complete staging for testing
This was not completed fully, however FASjson, IPA and Noggin are all deployed to staging but there are still some applications that are not in staging and using the new auth system yet.
* Import user data from FAS into IPA (Fedora)
There is still work being done on the script to be able to use this fully. Outstanding work is setting groups of users for fedora & centos and also being able to check for group conflicts and pushing them to noggin if there are not any.
* Pending successful run for Fedora, import data for CentOS users for testing matrix
No data was not imported, however there was testing done in a local dev environment and the matrix should work in stg once it is ready to be deployed when data is there
* Build confidence in the tooling used for reliancy when opening testing to the community in coming weeks
 Using a scale of 1-10, 1 being the lowest, the team feel we are at a 4 in confidence that we will be ready for community testing soon
* Create and update documentation on deployment & installation
Installation doc is in progress

### Sprint Goals
* Data import script completed
This will see the script able to dump information retrieved from FAS and restore it in IPA.
The script should be able to determine users from each group (CentOS + Fedora) and check for conflicts
All user and group data is imported to staging
** PR submitted and awaiting review - done

* Testing in staging for Noggin
This requires all fedora applications in staging so that they can be tested against the fasjson api
CentOS applications require user and group data in staging to be tested 
** This work is ongoing. There was a slight delay towards the end of the sprint due to planned reboot of stg (ticket https://pagure.io/fedora-infrastructure/issue/9439) but this work will continue through sprint 3 also.

* Tooling refinement 
Package fasjson-client as an rpm 
** Ongoing, some dependencies still need to be resolved.

* Queue Noggin for Security Audit
Architecture review
Installation documentation
** Installation documentation complete & architecture review will be submitted once testing in stg is more stable/complete.

* CentOS Noggin instance
This is a nice to have goal and not required this sprint 
** Will be moved into sprint 3.

### Sprint Goals
* Fedora Application testing completed in staging for ipsilon
** Use application table (link) to track testing against ipsilon
** Create a list of applications that require tests

* Data migration Script Completed
** PR review https://github.com/fedora-infra/fas2ipa/pull/20
** Send email to community lists explaining the merge and what the script will be used for to resolve multiple usernames and email address conflicts

* Communication Blasts!
** Create comms to CentOS on account merger
** Create comms on application testing requests for communtiy

* DNS entry for fasjson complete
** Required to be completed for community testing phase

* Develop a community testing phase plan
** CentOS 
*** Create ticket for CentOS users to access an rpm to test against (#47 in board)
*** Update ansible entry to point to data backend
*** Redirect openshift cluster to make sure users have access to the namespace
** Fedora
*** Log into application
*** Make sure user has right name and details
*** Users can log issues/bugs in the tracker once open
*** Agree on testing timeframe with team

* CentOS Login for Noggin
** Refactoring ansible playbook to include the redirect to a Noggin Portal for CentOS users when logging in
** There is some unknowns with certs that needs to be investigated
** Requires too instances that will pull from IPA 

### Sprint Goals
* Data Migration Script cleanup & testing
** Debugging user attributes
** auto populate group members 

* Mailman application integrated with Noggin/fasjson

* Check applications are authenticating with API

* Testing plan created for community announcement 
** details to be added

### Sprint Goals
* PO is absent from the meeting so team does not need to do anything.
* Remaining applications need ports to FASJSON
** Some have ports, but need to be pushed over the line/merged
** These then need to be deployed to staging to be tested
* Communication to the community about the current status of the new AAA system
**Testing plan to be created and shared with the community
* Security-related tickets need to be addressed before we move to production

**Comment by ryanlerch** [comment]
AAA is deployed.

going to close this one off
